### PR TITLE
C++: Additional pointer tests for DefaultTaintTracking.

### PR DESCRIPTION
**Pull request by geoffw0**
I believe better handling of these cases will restore much of the SAMATE regressions documented in https://github.com/github/codeql-c-analysis-team/issues/27.  Note that the AST DefaultTaintTracking never caught lines 254 or 256 (so coverage of those is probably not important) but it did flag the other tainted sinks (along with a few false positives).

### CODE REVIEW

**Comment by geoffw0** [comment]
@dbartol would you mind taking a look at these and tell me how feasible it is to cover any or all of the cases?  I haven't checked whether the issues are specific to DefaultTaintTracking, which probably matters.

The problem does not appear to be in models as I had said yesterday, sorry for any confusion - do you think we should remove or rephrase that potentially misleading comment about flow to output arguments:
```
TODO: This won't work in practice as long as all aliased memory is tracked together in a single virtual variable.
```

**Comment by MathiasVP** [comment]
Possible list of offenders of mine include:
- https://github.com/github/codeql/pull/4230
- https://github.com/github/codeql/pull/3933

I'll try a bisect and see if I can find the commit that broke these tests

File: cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/defaulttainttracking.cpp [hunk 1]
```diff
@@ -208,4 +208,50 @@ void test_field_to_obj_test_pointer_arith(Point* pp) {
...
+	ptr1 = buffer;
+	ptr2 = &ptr1;
+	memcpy(buffer, s, 1024);
+	ptr3 = buffer;
+	ptr4 = &ptr3;
+
+	sink(buffer); // tainted
+	sink(ptr1); // tainted
+	sink(ptr2);
+	sink(*ptr2); // tainted [NOT DETECTED]
```
Comment (MathiasVP) [inline cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/defaulttainttracking.cpp:231]: I think this sink (and the one on line 234) aren't caught is due to the two-level indirection (i.e., `ptr2` is a `char**`, and not just a `char*`).

**Comment by jbj** [comment]
> do you think we should remove or rephrase that potentially misleading comment about flow to output arguments:

Yes, let's remove it. I think there was agreement on Slack that this comment is out of date.

**Comment by geoffw0** [comment]
> Yes, let's remove it.

Done.

**Review by jbj — APPROVED** [review]

**Pull request merged**